presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewQueryOptimizer.java

```diff
@@ -349,6 +356,37 @@ protected Node visitFunctionCall(FunctionCall node, Void context)
                 rewrittenArguments.build());
     }
 
+    /**
+     * This is special-cased for now as COUNT is the only non-associative
+     * function supported by MV rewrites. In the future, we will want to
+     * support more non-associative functions and explore more
+     * extensible options.
+     *
+     * Functions in optimized MV queries are by default expanded to
+     * func(column_derived_from_func_in_mv). This only works for associative
+     * functions. Count is non-associative: COUNT(x U y) != COUNT(Count(x), COUNT(y)).
+     * Rather, COUNT(x U y) == SUM(COUNT(x), COUNT(y). This is what we do here.
+     * */
+    private FunctionCall rewriteCountAsSum(FunctionCall node, Expression derivedColumnName)
+    {
+        if (!node.getName().equals(COUNT)) {
+            throw new SemanticException(NOT_SUPPORTED, node, "Provided function was not COUNT");
+        }
+
+        if (node.isDistinct()) {
+            throw new SemanticException(NOT_SUPPORTED, node, "COUNT(DISTINCT) is not supported for materialized view query rewrite optimization");
+        }
```

Comment:
you can just directly use `checkArgument` helper

Reply:
That's what I originally did, but changed to address this comment:

https://github.com/prestodb/presto/pull/17842#discussion_r892678257

I like @pranjalssh's approach a little better because it's consistent with the rest of the file. I'm going to heavily refactor the file soon and would replace all of the raw `throw`s with `checkArgument`s then. WDYT?